base/platform/base_accessibility.cpp

```diff
@@ -0,0 +1,218 @@
+#include <QAccessibleWidget>
+
+#ifdef Q_OS_WIN
+#include <windows.h>
+#include <tlhelp32.h>
+#elif defined(Q_OS_MAC)
+#include <CoreFoundation/CoreFoundation.h>
+#else // For Linux, BSD, and other Unix-like systems
+#include <QProcess>
+#endif
+
+namespace base::Platform::Accessibility {
+
+	void ScreenReaderState::performCheck() {
+		crl::async([this] {
+			const auto isActive = detectScreenReader();
+			crl::on_main([this, isActive] {
+				handleResult(isActive);
+				});
+			});
+	}
+
+	bool ScreenReaderState::isScreenReaderActiveViaApi() {
+#ifdef Q_OS_WIN
+		BOOL screenReaderRunning = FALSE;
+		return (SystemParametersInfoW(SPI_GETSCREENREADER, 0, &screenReaderRunning, 0) && screenReaderRunning);
+#else
+		return false;
+#endif
+	}
+
+	bool ScreenReaderState::isScreenReaderProcessRunning() {
+#ifdef Q_OS_WIN
+		HANDLE snapshot = CreateToolhelp32Snapshot(TH32CS_SNAPPROCESS, 0);
+		if (snapshot == INVALID_HANDLE_VALUE) {
+			return false;
+		}
+
+		struct HandleCloser {
+			HANDLE handle;
+			~HandleCloser() { CloseHandle(handle); }
+		} closer = { snapshot };
+
+		PROCESSENTRY32W entry;
+		entry.dwSize = sizeof(PROCESSENTRY32W);
+		const WCHAR* screenReaders[] = {
+			L"nvda.exe", L"jaws.exe", L"jfw.exe",
+			L"windowseyes.exe", L"narrator.exe", L"dragon.exe"
+		};
+
+		if (Process32FirstW(snapshot, &entry)) {
+			do {
+				for (const auto& reader : screenReaders) {
+					if (_wcsicmp(entry.szExeFile, reader) == 0) {
+						return true;
+					}
+				}
+			} while (Process32NextW(snapshot, &entry));
+		}
+		return false;
+#else
+		return false;
+#endif
+	}
+
+	bool ScreenReaderState::detectForWindows() {
+		return isScreenReaderActiveViaApi() || isScreenReaderProcessRunning();
+	}
+
+	bool ScreenReaderState::detectForMac() {
+#ifdef Q_OS_MAC
+		Boolean keyExists = false;
+		Boolean voiceOverEnabled = CFPreferencesGetAppBooleanValue(
+			CFSTR("voiceOverOnOffKey"),
+			CFSTR("com.apple.universalaccess"),
+			&keyExists
+		);
+		return (keyExists && voiceOverEnabled);
+#else
+		return false;
+#endif
+	}
+
+	bool ScreenReaderState::detectForLinux() {
+#if !defined Q_OS_WIN && !defined Q_OS_MAC
+		QStringList envVars = { "AT_SPI_BUS", "AT_SPI_IOR", "GNOME_ACCESSIBILITY" };
+		for (const auto& var : envVars) {
+			if (!qgetenv(var.toLocal8Bit().constData()).isEmpty()) {
+				return true;
+			}
+		}
+		QStringList screenReaders = { "orca", "speakup", "brltty" };
+		for (const auto& reader : screenReaders) {
+			QProcess process;
+			process.start("pgrep", QStringList() << reader);
+			if (process.waitForFinished(1000) && process.exitCode() == 0) {
+				return true;
+			}
+		}
+		return false;
+#else
+		return false;
+#endif
+	}
+
+	bool ScreenReaderState::detectScreenReader() {
+#ifdef Q_OS_WIN
+		return detectForWindows();
+#elif defined(Q_OS_MAC)
+		return detectForMac();
+#else
+		return detectForLinux();
+#endif
+	}
+
+	namespace {
+		constexpr auto kRoleProperty = "_a11y_custom_role";
+		constexpr auto kNameProperty = "_a11y_custom_name";
```

Comment:
By using RpWidget, we can add these values to native code rather than Qt property which will be more perfomant